packages/roosterjs-editor-core/lib/coreApi/selectTable.ts

```diff
@@ -0,0 +1,206 @@
+import {
+    EditorCore,
+    SelectionRangeTypes,
+    TableSelection,
+    SelectTable,
+} from 'roosterjs-editor-types';
+
+const TABLE_ID = 'tableSelected';
+const CONTENT_DIV_ID = 'contentDiv_';
+
+let styleElement: HTMLStyleElement;
+
+/**
+ * @internal
+ * Select a table and save data of the selected range
+ * @param core The EditorCore object
+ * @param table table to select
+ * @param coordinates first and last cell of the selection, if this parameter is null, instead of
+ * selecting, will unselect the table.
+ * @returns true if successful
+ */
+export const selectTable: SelectTable = (
+    core: EditorCore,
+    table: HTMLTableElement,
+    coordinates?: TableSelection
+) => {
+    const doc = core.contentDiv.ownerDocument;
+
+    if (coordinates) {
```

Comment:
When table or cooridinates is null, we can just unselect then exist. So that we can always unselect then reselect from Editor class, and in this core API we just need to do either select or unselect